rest/route/github_mock.go

```diff
@@ -0,0 +1,218 @@
+package route
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/commitqueue"
+	"github.com/evergreen-ci/evergreen/rest/data"
+	restModel "github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/utility"
+	"github.com/google/go-github/v34/github"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+type mockGithubHookApi struct {
+	queue  amboy.Queue
+	secret []byte
+
+	event     interface{}
+	eventType string
+	msgID     string
+	sc        data.Connector
+	settings  *evergreen.Settings
+}
+
+func makeMockGithubHooksRoute(sc data.Connector, queue amboy.Queue, secret []byte, settings *evergreen.Settings) gimlet.RouteHandler {
```

Comment:
Similarly, instead of mocking the route itself, I think we need a small mock connector

Reply:
Mock connector is added, removed this now